app/src/main/java/com/github/se/eventradar/model/repository/event/FirebaseEventRepository.kt

```diff
@@ -118,4 +120,65 @@ class FirebaseEventRepository(db: FirebaseFirestore = Firebase.firestore) : IEve
 
     awaitClose { listener.remove() }
   }
+
+  override suspend fun addAttendee(eventId: String, attendeeUserId: String): Resource<Unit> {
+    return try {
+      eventRef.document(eventId).update("attendees_list", FieldValue.arrayUnion(attendeeUserId))
+      Resource.Success(Unit)
+    } catch (e: Exception) {
+      Resource.Failure(e)
+    }
+  }
+
+  override suspend fun removeAttendee(eventId: String, attendeeUserId: String): Resource<Unit> {
+    return try {
+      eventRef.document(eventId).update("attendees_list", FieldValue.arrayRemove(attendeeUserId))
+      Resource.Success(Unit)
+    } catch (e: Exception) {
+      Resource.Failure(e)
+    }
+  }
+
+  override suspend fun incrementPurchases(eventId: String): Resource<Unit> {
+    try {
+      db.runTransaction { transaction ->
+        val ref = eventRef.document(eventId)
+        val resultDocument = transaction.get(ref)
+        val purchases = resultDocument.getLong("ticket_purchases")
+        val capacity = resultDocument.getLong("ticket_capacity")
+
+        if (purchases != null && capacity != null) {
+          if (purchases < capacity) {
+            transaction.update(ref, "ticket_purchases", purchases + 1)
+          } else {
+            throw FirebaseFirestoreException(
+                "No more available tickets",
+                FirebaseFirestoreException.Code.ABORTED,
+            )
+          }
+        } else {
+          throw FirebaseFirestoreException(
+              "Invalid data",
+              FirebaseFirestoreException.Code.ABORTED,
+          )
+        }
+      }
+      return Resource.Success(Unit)
+    } catch (e: Exception) {
+      return Resource.Failure(e)
+    }
+  }
+
+  override suspend fun decrementPurchases(eventId: String): Resource<Unit> {
+    return addSubPurchases(eventId, -1)
+  }
+
+  private fun addSubPurchases(eventId: String, value: Long): Resource<Unit> {
+    return try {
+      eventRef.document(eventId).update("ticket_purchases", FieldValue.increment(value))
```

Comment:
maybe it's also useful to add a check here to ensure that the ticket_purchases value does not become less than zero and throw an error if it does :)